File: TelerikUIExamples/TelerikUIExamples/DateTimeCategoryAxis.m
//
//  DateTimeCategoryAxis.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import <TelerikUI/TelerikUI.h>
#import "DateTimeCategoryAxis.h"
#import "StockDataPoint.h"

@interface DateTimeCategoryAxis ()
@end

@implementation DateTimeCategoryAxis

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    TKChart *chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:chart];
    
    
    TKChartDateTimeCategoryAxis *xAxis = [TKChartDateTimeCategoryAxis new];
    NSDateFormatter *formatter = [NSDateFormatter new];
    formatter.dateFormat = @"d MMM";
    xAxis.labelFormatter = formatter;
    xAxis.style.majorTickStyle.ticksOffset = -3;
    xAxis.style.majorTickStyle.ticksHidden = NO;
    xAxis.style.majorTickStyle.ticksWidth = 1.5;
    xAxis.style.majorTickStyle.ticksFill = [TKSolidFill solidFillWithColor:[UIColor colorWithRed:203/255.0 green:203/255.0 blue:203/255.0 alpha:1.0]];
    xAxis.style.majorTickStyle.maxTickClippingMode = TKChartAxisClippingModeVisible;
    xAxis.plotMode = TKChartAxisPlotModeBetweenTicks;
    
    TKChartNumericAxis *yAxis = [[TKChartNumericAxis alloc] initWithMinimum:@320 andMaximum:@360];
    yAxis.majorTickInterval = @20;
    
    TKChartCandlestickSeries *series = [[TKChartCandlestickSeries alloc] initWithItems:[StockDataPoint stockPoints:10]];
    series.gapLength = 0.6;
    series.xAxis = xAxis;
    chart.yAxis = yAxis;

    [chart addSeries:series];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Examples/Examples/Chart/AxisTypes/DateTimeCategoryAxis.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("DateTimeCategoryAxis")]
	public class DateTimeCategoryAxis: XamarinExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			// >> chart-axis-datetime-cat-cs
			var xAxis = new TKChartDateTimeCategoryAxis ();
			var formatter = new NSDateFormatter ();
			formatter.DateFormat = "d MMM";
			xAxis.LabelFormatter = formatter;
			// << chart-axis-datetime-cat-cs

			// >> chart-tick-style-set-cs
			xAxis.Style.MajorTickStyle.TicksOffset = -3;
			xAxis.Style.MajorTickStyle.TicksHidden = false;
			xAxis.Style.MajorTickStyle.TicksWidth = 1.5f;
			xAxis.Style.MajorTickStyle.TicksFill = new TKSolidFill (new UIColor (203 / 255.0f, 203 / 255.0f, 203 / 255.0f, 1f));
			xAxis.Style.MajorTickStyle.MaxTickClippingMode = TKChartAxisClippingMode.Visible;
			// << chart-tick-style-set-cs

			// >> chart-category-plot-cs
			xAxis.PlotMode = TKChartAxisPlotMode.BetweenTicks;
			// << chart-category-plot-cs

			var yAxis = new TKChartNumericAxis (new NSNumber(320), new NSNumber(360));
			yAxis.MajorTickInterval = 20;

			var series = new TKChartCandlestickSeries(StockDataPoint.LoadStockPoints(10).ToArray());
			// >> chart-gap-cs
			series.GapLength = 0.6f;
			// << chart-gap-cs
			series.XAxis = xAxis;
			chart.YAxis = yAxis;

			chart.AddSeries (series);
		}
	}
}

